tests/system/feature_store/test_spark_engine.py

```diff
@@ -1524,6 +1524,73 @@ def test_ingest_and_get_offline_features_with_hdfs(self):
 
         resp_df.reset_index(drop=True, inplace=True)
         pd.testing.assert_frame_equal(resp_df[["bad", "department"]], expected_df)
+        target.purge()
+        with pytest.raises(FileNotFoundError):
+            target.as_df()
+        # Checks that a FileNotFoundError is not raised in the rm method.
+        target.purge()
+
+    @pytest.mark.skipif(
+        not {"HDFS_HOST", "HDFS_PORT", "HDFS_HTTP_PORT", "HADOOP_USER_NAME"}.issubset(
+            os.environ.keys()
+        ),
+        reason="HDFS host, ports and user name are not defined",
+    )
+    def test_hdfs_wrong_credentials(self):
+        datastore_profile = DatastoreProfileHdfs(
+            name="my-hdfs",
+            host=os.getenv("HDFS_HOST"),
+            port=int(os.getenv("HDFS_PORT")),
+            http_port=int(os.getenv("HDFS_HTTP_PORT")),
+            user="wrong-user",
+        )
+        register_temporary_client_datastore_profile(datastore_profile)
+        self.project.register_datastore_profile(datastore_profile)
+        target = ParquetTarget(
+            "mytarget", path=f"{self.hdfs_output_dir}-get_offline_features"
+        )
+        with pytest.raises(PermissionError):
+            target.purge()
+
+    @pytest.mark.skipif(
+        not {"HDFS_HOST", "HDFS_PORT", "HDFS_HTTP_PORT", "HADOOP_USER_NAME"}.issubset(
+            os.environ.keys()
+        ),
+        reason="HDFS host, ports and user name are not defined",
+    )
+    def test_hdfs_empty_host(self):
+        host = os.environ.pop("HDFS_HOST")
+        port = os.environ.pop("HDFS_PORT")
+        http_port = os.environ.pop("HDFS_HTTP_PORT")
```

Comment:
I think you can do one better and use `monkeypatch.setenv` so you don't need the `finally` clause at all. See examples in the code base.